packages/core/src/redocly/index.ts

```diff
@@ -1,58 +1,78 @@
 import { existsSync, readFileSync, writeFileSync, unlinkSync } from 'fs';
 import { resolve } from 'path';
 import { homedir } from 'os';
-import { yellow, red, green, gray } from 'colorette';
+import { red, green, gray } from 'colorette';
 import { RegistryApi } from './registry-api';
+import { DOMAINS, Region } from '../config/config';
 
 const TOKEN_FILENAME = '.redocly-config.json';
+const DEFAULT_REGION = 'us';
+const DEFAULT_DOMAIN = 'redoc.ly';
 
 export class RedoclyClient {
-  private accessToken: string | undefined;
+  private accessTokens: { us?: string; eu?: string; } | undefined;
+  private domain: string | undefined;
+  private region: Region;
   registryApi: RegistryApi;
 
-  constructor() {
-    this.loadToken();
-    this.registryApi = new RegistryApi(this.accessToken);
+  constructor(region?: Region) {
+    this.region = region || DEFAULT_REGION;
+    this.loadTokens();
+    this.setDomain(region);
+    this.registryApi = new RegistryApi(this.getTokenByRegion(), this.domain);
   }
 
-  hasToken(): boolean {
-    return !!this.accessToken;
+  getTokenByRegion() {
+    return this.accessTokens && this.accessTokens[this.region];
   }
 
-  loadToken(): void {
+  setDomain(region?: Region) {
+    this.domain = region
+      ? DOMAINS[region]
+      : process.env.REDOCLY_DOMAIN || DOMAINS[DEFAULT_REGION];
+  }
+
+  getDomain() {
+    return this.domain || DEFAULT_DOMAIN;
+  }
+
+  hasTokens(): boolean {
+    return !!this.accessTokens;
+  }
+
+  loadTokens(): void {
     if (process.env.REDOCLY_AUTHORIZATION) {
-      this.accessToken = process.env.REDOCLY_AUTHORIZATION;
+      this.accessTokens = {};
+      for (const domain in DOMAINS) {
+        this.accessTokens[domain as Region] = process.env.REDOCLY_AUTHORIZATION;
+      }
```

Comment:
I thought a bit more about it. Let's use this token not for all domains but just for the current one: `this.domain`.

I believe this makes more sense.

```suggestion
      this.accessTokens[this.domain] = process.env.REDOCLY_AUTHORIZATION;
```